### ISSUE TITLE
Products Base - Start Rating Alignment Issue (Elementor Pro)

### ISSUE THREAD

**Issue by amyriounis** [elementor/elementor#20920]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

In `products-base.php` the Alignment option in the style tab is defined as such:
![DeepinScreenshot_select-area_20230104173242](https://user-images.githubusercontent.com/9055437/210613614-8db0a482-aee6-49ae-a3ef-1bb5d003b8bf.png)

The Alignment option in the editor is this one:
![DeepinScreenshot_select-area_20230104173328](https://user-images.githubusercontent.com/9055437/210614242-1cc3067c-d0f2-487c-93d9-07b53e7bebd5.png) 

The problem stems from the fact that the `prefix_class` value is missing the `%s` identifier which is used for outputting the device class names in the frontend. As such, the responsive values for this option cannot work properly as they are all applied at the same time like so:
![;oi](https://user-images.githubusercontent.com/9055437/210616645-f3af3751-1ca0-42d2-a2f0-51aeef2fcf40.png) 

### Steps to reproduce

1 - Have products with ratings.
2 - Add a Products / Archive Products / Product Related / Upsells widget 
3 - Set different values for the Alignment option under Style Tab->Products->Alignment.
4 - The ratings will not align properly for the star rating. 

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

Elementor v3.9.2
Elementor Pro v3.9.1

**Comment by nicholaszein** [comment]
Hello @amyriounis! 

Thank you for submitting your issue and for the detailed explanation! 🙏 

🏷️ This issue has been labeled and will be forwarded to the team for validation and possible solution. Please keep in mind that we can't provide any ETA on when this issue could be fixed if it is validated. We ask for patience.

Best regards

**Comment by nicholaszein** [comment]
Hello @amyriounis, 

Thank you for submitting your report! 🙏 

### ❌ While reviewing your report, we noticed it was missing some key pieces of information:

- 💾 **System info:** Please include the System Info report as outlined in our [guidelines](https://github.com/elementor/elementor/issues/new?assignees=&labels=status%2Fawaiting_triage&template=bug.yml). 
   - Get your System Info from your WordPress `Dashboard` > `Elementor` > `System info` [learn how to copy it here](https://elementor.com/help/hare-my-system-info/). We can't accept issues without it.

Without them, we simply can't help. 😕 

### 🧐 Please provide the missing information above so we can troubleshoot your issue faster and better.

Best regards

**Comment by Voltaic-Coffee** [comment]
# tmp
I have experienced the same issue and will provide the system info from my setup, which uses a different theme.


<details>
<summary>System Info</summary>

== Server Environment ==
	Operating System: Linux
	Software: nginx/1.24.0
	MySQL version: (Google) v8.0.31
	PHP Version: 8.2.16
	PHP Memory Limit: 1024M
	PHP Max Input Vars: 10000
	PHP Max Post Size: 256M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.7.1
	Site URL: https://example.com
	Home URL: https://example.com
	WP Multisite: No
	Max Upload Size: 256 MB
	Memory limit: 1024M
	Max Memory limit: 1024M
	Permalink Structure: /%postname%/
	Language: en_US
	Timezone: America/New_York
	Admin Email: email@example.com
	Debug Mode: Inactive

== Theme ==
	Name: Astra
	Version: 4.8.8
	Author: Brainstorm Force
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en-US
	User Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36

== Active Plugins ==
	Age Verification
		Version: 2.9.5
		Author: Devio Digital

	All-in-One WP Migration
		Version: 7.87
		Author: ServMask

	All-in-One WP Migration Unlimited Extension
		Version: 2.64
		Author: ServMask

	CartPops
		Version: 1.5.33
		Author: CartPops.com

	Customizer for WooCommerce
		Version: 2.9.0
		Author: SkyVerge

	Elementor
		Version: 3.26.2
		Author: Elementor.com

	Elementor Pro
		Version: 3.26.1
		Author: Elementor.com

	ElementsKit Lite
		Version: 3.3.4
		Author: Wpmet

	ElementsKit Pro
		Version: 3.7.1
		Author: Wpmet

	LiteSpeed Cache
		Version: 6.5.4
		Author: LiteSpeed Technologies

	Migrate Guru
		Version: 5.88
		Author: Migrate Guru

	Product Filter by WBW
		Version: 2.7.6
		Author: woobewoo

	Product Filter Widget for Elementor
		Version: 1.0.4
		Author: QualArch

	WooCommerce
		Version: 9.5.1
		Author: Automattic

	WooCommerce Authorize.Net Gateway
		Version: 3.10.9
		Author: SkyVerge

	Wordfence Security
		Version: 8.0.1
		Author: Wordfence

	WPC Buy Now Button for WooCommerce
		Version: 2.1.1
		Author: WPClever


== Elements Usage ==
	
	container : 12
		button : 6
		container : 46
		heading : 15
		image : 3
		text-editor : 7
	page : 1
		button : 3
		container : 12
		heading : 5
		shortcode : 2
		text-editor : 3
	wp-page : 18
		button : 40
		container : 258
		elementskit-image-box : 2
		heading : 123
		icon : 1
		icon-box : 4
		image : 51
		shortcode : 16
		slides : 1
		spacer : 20
		testimonial-carousel : 1
		text-editor : 50


== Settings ==
	
	cpt_support: post, page, elementskit_content, elementskit_template, product
	allow_tracking: yes
	unfiltered_files_upload: 1
	font_display: swap


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	
	woocommerce: Active


== Elementor Experiments ==
	Inline Font Icons: Active by default
	Additional Custom Breakpoints: Active by default
	Container: Active by default
	Upgrade Swiper Library: Active by default
	Nested Elements Performance: Active by default
	Optimized Markup: Inactive by default
	Plugin Onboarding: Active by default
	CSS Smooth Scroll: Active by default
	Head Loading Styles: Inactive by default
	Default to New Theme Builder: Active by default
	Elementor Home Screen: Inactive
	Landing Pages: Inactive by default
	Nested Elements: Active by default
	Editor Top Bar: Active by default
	Pages Panel: Inactive by default
	Element Caching: Active
	Link In Bio: Active by default
	Floating Buttons: Active by default
	Elementor Editor Events: Active
	Atomic Widgets: Inactive by default
	Global Classes: Inactive by default
	Launchpad Checklist: Inactive by default
	Improved Page Actions Layout: Inactive by default
	Menu: Inactive by default


== Log ==
	
Log: showing 20 of 302024-12-11 06:03:17 [info] Elementor/Upgrades - _on_each_version Start  
2024-12-11 06:03:18 [info] Elementor/Upgrades - _on_each_version Finished 
2024-12-11 06:03:18 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.10',
  'to' => '3.25.11',
)]
2024-12-16 16:32:09 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.11',
  'to' => '3.26.0',
)]
2024-12-16 16:32:10 [info] elementor::elementor_updater Started 
2024-12-16 16:32:10 [info] Elementor/Upgrades - _on_each_version Start  
2024-12-16 16:32:11 [info] Elementor/Upgrades - _on_each_version Finished 
2024-12-16 16:32:11 [info] Elementor/Upgrades - _v_3_26_0 Start  
2024-12-16 16:32:12 [info] Elementor/Upgrades - _v_3_26_0 Finished 
2024-12-16 16:32:12 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.11',
  'to' => '3.26.0',
)]
2024-12-19 16:35:44 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.26.0',
  'to' => '3.26.2',
)]
2024-12-19 19:52:49 [info] elementor::elementor_updater Started 
2024-12-19 19:52:49 [info] Elementor/Upgrades - _on_each_version Start  
2024-12-19 19:52:50 [info] Elementor/Upgrades - _on_each_version Finished 
2024-12-19 19:52:50 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.26.0',
  'to' => '3.26.2',
)]
2024-12-20 01:23:36 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.21.2',
  'to' => '3.26.1',
)]
2024-12-20 01:35:55 [info] elementor-pro::elementor_pro_updater Started 
2024-12-20 01:35:55 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-12-20 01:35:56 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-12-20 01:35:56 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.21.2',
  'to' => '3.26.1',
)]

PHP: showing 9 of 9PHP: 2024-11-25 08:58:18 [warning X 88][/mnt/shared-stack/plugins/elementor/3.25.10/includes/conditions.php::87] Undefined array key &quot;background_image&quot; [array (
  'trace' => '
#0: /mnt/shared-stack/plugins/elementor/3.25.10/includes/conditions.php(87): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: /mnt/shared-stack/plugins/elementor/3.25.10/includes/base/controls-stack.php(1438): Elementor\Conditions :: check()
#2: /mnt/shared-stack/plugins/elementor/3.25.10/includes/base/controls-stack.php(1222): Elementor\Controls_Stack -> is_control_visible()
#3: /mnt/shared-stack/plugins/elementor/3.25.10/includes/base/controls-stack.php(1227): Elementor\Controls_Stack -> get_active_settings()
#4: /mnt/shared-stack/plugins/elementor/3.25.10/includes/base/controls-stack.php(1263): Elementor\Controls_Stack -> get_active_settings()
',
)]
PHP: 2024-11-25 08:58:18 [warning X 88][/mnt/shared-stack/plugins/elementor/3.25.10/includes/conditions.php::90] Trying to access array offset on value of type null [array (
  'trace' => '
#0: /mnt/shared-stack/plugins/elementor/3.25.10/includes/conditions.php(90): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: /mnt/shared-stack/plugins/elementor/3.25.10/includes/base/controls-stack.php(1438): Elementor\Conditions :: check()
#2: /mnt/shared-stack/plugins/elementor/3.25.10/includes/base/controls-stack.php(1222): Elementor\Controls_Stack -> is_control_visible()
#3: /mnt/shared-stack/plugins/elementor/3.25.10/includes/base/controls-stack.php(1227): Elementor\Controls_Stack -> get_active_settings()
#4: /mnt/shared-stack/plugins/elementor/3.25.10/includes/base/controls-stack.php(1263): Elementor\Controls_Stack -> get_active_settings()
',
)]
PHP: 2024-11-25 22:33:53 [warning X 9][/mnt/shared-stack/plugins/elementor/3.25.10/modules/ai/connect/ai.php::693] Undefined array key &quot;mega-menu&quot; [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-11-25 23:22:58 [warning X 5][/mnt/shared-stack/plugins/elementor/3.25.10/includes/managers/image.php::113] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-11-26 03:33:52 [warning X 1][/mnt/shared-stack/plugins/elementor/3.25.10/core/files/file-types/svg.php::161] simplexml_load_file(): I/O warning : failed to load external entity &quot;https://example.com/wp-content/uploads/2024/11/Instagram_Fill.svg&quot; [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-12-12 11:10:21 [warning X 12][/mnt/shared-stack/plugins/elementor/3.25.11/includes/conditions.php::87] Undefined array key &quot;background_image&quot; [array (
  'trace' => '
#0: /mnt/shared-stack/plugins/elementor/3.25.11/includes/conditions.php(87): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: /mnt/shared-stack/plugins/elementor/3.25.11/includes/base/controls-stack.php(1438): Elementor\Conditions :: check()
#2: /mnt/shared-stack/plugins/elementor/3.25.11/includes/base/controls-stack.php(1222): Elementor\Controls_Stack -> is_control_visible()
#3: /mnt/shared-stack/plugins/elementor/3.25.11/includes/base/controls-stack.php(1227): Elementor\Controls_Stack -> get_active_settings()
#4: /mnt/shared-stack/plugins/elementor/3.25.11/includes/base/controls-stack.php(1263): Elementor\Controls_Stack -> get_active_settings()
',
)]
PHP: 2024-12-12 11:10:21 [warning X 12][/mnt/shared-stack/plugins/elementor/3.25.11/includes/conditions.php::90] Trying to access array offset on value of type null [array (
  'trace' => '
#0: /mnt/shared-stack/plugins/elementor/3.25.11/includes/conditions.php(90): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: /mnt/shared-stack/plugins/elementor/3.25.11/includes/base/controls-stack.php(1438): Elementor\Conditions :: check()
#2: /mnt/shared-stack/plugins/elementor/3.25.11/includes/base/controls-stack.php(1222): Elementor\Controls_Stack -> is_control_visible()
#3: /mnt/shared-stack/plugins/elementor/3.25.11/includes/base/controls-stack.php(1227): Elementor\Controls_Stack -> get_active_settings()
#4: /mnt/shared-stack/plugins/elementor/3.25.11/includes/base/controls-stack.php(1263): Elementor\Controls_Stack -> get_active_settings()
',
)]
PHP: 2024-12-19 20:25:02 [warning X 2][/mnt/shared-stack/plugins/elementor/3.26.2/includes/conditions.php::90] Trying to access array offset on value of type null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-12-20 01:50:06 [error X 1][/mnt/shared-stack/plugins/elementor/3.26.2/modules/ai/module.php::231] Uncaught Error: Call to a member function get_gallery_image_ids() on bool in /mnt/shared-stack/plugins/elementor/3.26.2/modules/ai/module.php:231
Stack trace:
#0 /mnt/customers/customers-bh-123abc/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/wp-content/wp-includes/class-wp-hook.php(324): Elementor\Modules\Ai\Module-&gt;get_product_images_ajax()
#1 /mnt/customers/customers-bh-123abc/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/wp-content/wp-includes/class-wp-hook.php(348): WP_Hook-&gt;apply_filters()
#2 /mnt/customers/customers-bh-123abc/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/wp-content/wp-includes/plugin.php(517): WP_Hook-&gt;do_action()
#3 /mnt/customers/customers-bh-123abc/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/wp-content/wp-admin/admin-ajax.php(192): do_action()
#4 {main}
  thrown [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 4 of 4JS: 2024-11-25 09:01:37 [error X 8][https://example.com/wp-admin/load-scripts.php?c=1&amp;loadchunk_0=jquery-core,jquery-migrate,wp-hooks&amp;ver=6.7.1:2:28760] Cannot read properties of undefined (reading &#039;value&#039;) 
JS: 2024-11-25 23:51:18 [error X 5][https://example.com/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=3.25.10:2:12921] Cannot read properties of undefined (reading &#039;attributes&#039;) 
JS: 2024-12-12 22:21:47 [error X 1][https://example.com/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.25.11:3:54583] Cannot read properties of undefined (reading &#039;activate&#039;) 
JS: 2024-12-20 00:47:25 [error X 1][https://example.com/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.26.2:3:801849] Cannot read properties of undefined (reading &#039;localized&#039;) 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible
	Product Filter Widget for Elementor: Compatibility not specified

== Elementor Pro - Compatibility Tag ==
	
</details>
